test/libsolidity/SemanticTest.cpp

```diff
@@ -300,7 +301,11 @@ TestCase::TestResult SemanticTest::runTest(
 		}
 
 		for (std::unique_ptr<TestHook> const& hook: m_testHooks)
-			hook->afterFunctionCall(test);
+		{
+			vector<string> reactions = hook->afterFunctionCall(test);
```

Comment:
Can't we use a specific function just for the reactions/effects?

Reply:
I think that would be possible. So you prefer to have next to `Builtins` and `TestHooks` a third mechanism called e.g. `Effects`?

Reply:
I mean, if I understood you correctly, you want to decouple this mechanism from the `TestHook`'s right?

Reply:
Yes